mso/utils.go

```diff
@@ -323,3 +323,107 @@ func duplicatesInList(list []string) []string {
 	}
 	return duplicates
 }
+
+func GetTemplateIdFromResourceId(input string) (string, error) {
+	parts := strings.Split(input, "/")
+	if parts[0] != "templateId" {
+		return "", fmt.Errorf("invalid id provided")
+	}
+	return parts[1], nil
+}
+
+func GetPolicyNameFromResourceId(input, policyType string) (string, error) {
+	parts := strings.Split(input, "/")
+
+	for i := 0; i < len(parts)-1; i++ {
+		if parts[i] == policyType {
+			if i+1 < len(parts) {
+				return parts[i+1], nil
+			}
+			return "", fmt.Errorf("no value found after policyType")
+		}
+	}
+
+	return "", fmt.Errorf("policyType not found in the id")
+}
+
+func GetPolicyIndexByName(cont *container.Container, policyName string, templateElements ...string) (int, error) {
```

Comment:
`GetPolicyIndexByName` and `GetPolicyIndexByUuid` look similar; is it possible to merge them?

Can we use `searchKey` and `searchValue` to match the object?

Reply:
I think I would agree with Sabari his comment, because what if the value is not name but ip for instance? Or what if the identifying part of the policy is multiple keys? Then we would need add another function. We could wait until we hit that case and change then.